index.js

```diff
@@ -373,25 +373,115 @@ module.exports = function (log, indexesPath) {
         : Buffer.from(op.data.value)
   }
 
-  function ensureOffsetIndexSync(cb) {
-    if (log.since.value > indexes['offset'].seq)
-      updateIndex({ data: { indexName: 'offset' } }, cb)
+  function ensureIndexSync(opOrName, cb) {
```

Comment:
I think I would prefer this to be ensureIndexSync as it was before, and then a helper function to generate an "fake" op from name.

Reply:
the function was a bit hard for me to read. It looks as if it does a bunch of things, but mainly just mangles stuff into the right shape

Reply:
Fine by me, I'll do that